### DISCUSSION TITLE
Help: No netflow-data received

### DISCUSSION THREAD

**Discussion by SamLowrie** [akvorado/akvorado#267] [Q&A]
I have a hard time to get akvorado to consume any netflows.

I installed the docker container on a fresh Debian 11 installation with docker-compose. The containers spin up when started.

The netflow originates from an ASR9901, configuration is similar to the akvorado documentation example. I have successfully captured netflow data on the local system on port 2055. Netflow is setup to send templates at least every 60 seconds. A UDP-inlet netflow is configured under flow/inputs on port 2055 and a open port is visible when using netstat.

However the trail ends here. Using ‚`curl -s 'http://127.0.0.1:8080/api/v0/inlet/flows?limit=1` connects to the server, but never returns any data. The command `curl -s 'http://127.0.0.1:8080/api/v0/inlet/metrics' | grep '^akvorado_inlet_flow_input_udp_packets`does not return any entry.

One of the first erros in the logfile is:
```
akvorado-inlet_1            | Error: unable to fetch configuration file: Get "http://akvorado-orchestrator:8080/api/v0/orchestrator/configuration/inlet": dial tcp 172.18.0.4:8080: connect: connection refused
akvorado-orchestrator_1     | {"level":"error","error":"dial tcp 172.18.0.10:9000: connect: connection refused","time":"2022-11-22T12:50:26Z","caller":"akvorado/orchestrator/clickhouse/migrations.go:48","module":"akvorado/orchestrator/clickhouse","message":"unable to query database"}
```

The `akvorado-orchestrator_1` error repeats many times during startup. Is this the problem? What to do about it?

Version-Info:
```
akvorado v1.6.2
  Build date: 2022-11-03T21:10:15+0000
  Built with: go1.19.2
```

docker ps:
```
CONTAINER ID   IMAGE                               COMMAND                  CREATED        STATUS                    PORTS                                                                                NAMES
5c851a937d85   bitnami/kafka:2.8.1                 "/opt/bitnami/scriptâ¦"   20 hours ago   Up 13 minutes (healthy)   9092/tcp                                                                            akvorado_kafka_1
3f4764282a15   bitnami/zookeeper:3.6               "/opt/bitnami/scriptâ¦"   20 hours ago   Up 13 minutes             2181/tcp,2888/tcp, 3888/tcp, 8080/tcp                                               akvorado_zookeeper_1
448ee161c9d1   traefik:2.6                         "/entrypoint.sh --apâ¦"   20 hours ago   Up 13 minutes             80/tcp, 127.0.0.1:8080->8080/tcp, 0.0.0.0:8081->8081/tcp                             akvorado_traefik_1
2b4c880870c3   clickhouse/clickhouse-server:22.8   "/entrypoint.sh"         20 hours ago   Up 13 minutes (healthy)   8123/tcp, 9000/tcp, 9009/tcp                                                         akvorado_clickhouse_1
70a1fc28a9dc   ghcr.io/akvorado/akvorado:latest    "/usr/local/bin/akvoâ¦"   20 hours ago   Up 12 minutes (healthy)   0.0.0.0:2055->2055/udp, 0.0.0.0:6343->6343/udp, 8080/tcp, 0.0.0.0:10179->10179/tcp   akvorado_akvorado-inlet_1
511e70dca7a2   ghcr.io/akvorado/akvorado:latest    "/usr/local/bin/akvoâ¦"   20 hours ago   Up 13 minutes (healthy)   8080/tcp                                                                             akvorado_akvorado-console_1
5013f5be8869   provectuslabs/kafka-ui:v0.4.0       "/bin/sh -c 'java $Jâ¦"   20 hours ago   Up 13 minutes             8080/tcp                                                                             akvorado_kafka-ui_1
7ceb32ce6c11   ghcr.io/akvorado/akvorado:latest    "/usr/local/bin/akvoâ¦"   20 hours ago   Up 12 minutes (healthy)   8080/tcp                                                                             akvorado_akvorado-orchestrator_1
2dc3b99366a7   ghcr.io/akvorado/akvorado:latest    "/usr/local/bin/akvoâ¦"   20 hours ago   Up 13 minutes                                                                                                  akvorado_akvorado-conntrack-fixer_1
```

**Comment by vincentbernat** [comment]
The first errors don't matter much as Docker spawns stuff in random order. Not everything is ready when it needs to be, so containers are restarted until the situation is stable. This is your case.

Since you don't receive anything, check that you get metrics without the `grep`. If yes, check conntrack entries for port 2055:

```
conntrack -L -p udp --orig-port-dst 2055
```

**Reply by SamLowrie** [reply]
Thanks for the clarification of the startup process and your suggestions.

Your suggestions didn't work so i tried to dig deeper ... and I finally found the problem:

**app_armor was active and blocked (silently) several of the docker processes!**

This probably might occur to other Debian 11 users as well. After disabling app_armor the inlet-service started to receive flows.
